Simple Inventory Management System/Inventory.cs

```diff
@@ -28,5 +28,40 @@ public void DisplayProducts()
             }
         }
 
+        public void EditProduct(string productName)
+        {
+            Product product = products.Find(p => p.Name == productName);
+            if (product != null)
+            {
+                Console.WriteLine($"Product found: {product.Name} - Price: ${product.Price} - Quantity: {product.Quantity}");
+                Console.WriteLine("Enter new details:");
+
+                Console.Write("New name (leave blank to keep unchanged): ");
+                string newName = Console.ReadLine();
+                if (!string.IsNullOrEmpty(newName))
+                {
+                    product.Name = newName;
+                }
+
+                Console.Write("New price : ");
+                string newPriceStr = Console.ReadLine();
+                if (!string.IsNullOrEmpty(newPriceStr))
+                {
+                    decimal newPrice;
+                    if (decimal.TryParse(newPriceStr, out newPrice))
+                    {
+                        product.Price = newPrice;
+                    }
+                    else
+                    {
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        Console.WriteLine("Invalid price format. Price not updated.");
+                        Console.ForegroundColor = ConsoleColor.White;
+                    }
+                }
+
+            }
+        }
```

Comment:
The method currently does multiple things: finds the product, displays its details, and handles user input for editing.  Try to split it into multiple methods.